tests/FunctionalConnectorTest.php

```diff
@@ -29,4 +34,174 @@ public function connectionToTcpServerShouldSucceedWithLocalhost()
         $connection->close();
         $server->close();
     }
+
+    /**
+     * @test
+     * @group internet
+     */
+    public function connectionToRemoteTCP4n6ServerShouldResultInOurIP()
+    {
+        $loop = Factory::create();
+
+        $connector = new Connector($loop, array('happy_eyeballs' => true));
+
+        $ip = Block\await($this->request('dual.tlund.se', $connector), $loop, self::TIMEOUT);
+
+        $this->assertSame($ip, filter_var($ip, FILTER_VALIDATE_IP, FILTER_FLAG_IPV4 | FILTER_FLAG_IPV6), $ip);
+    }
+
+    /**
+     * @test
+     * @group internet
+     */
+    public function connectionToRemoteTCP4ServerShouldResultInOurIP()
+    {
+        if ($this->ipv4() === false) {
+            // IPv4 not supported on this system
+            $this->assertFalse($this->ipv4());
+            return;
```

Comment:
And reason why the test is not marked as skipped instead? Doing an assertion makes no sense to me. Same below with IPv6.

Reply:
@clue I agree with @CharlotteDunois, marking it skipped makes more sense and is IMHO cleaner